lxqt-sudo/man/lxqt-sudo.1
.TH lxqt-sudo 1 "" "" "LXQt\ Module"
.SH NAME
\fBlxqt-sudo\fR, \fBlxsu\fR \- execute a command as privileged user
.SH SYNOPSIS
\fBlxqt-sudo\fR \fIoption\fR [\fIcommand\fR [\fIarguments\fR]]
.br
\fBlxsu\fR [\fIoption\fR] [\fIcommand\fR [\fIarguments\fR]]
.br
\fBlxsudo\fR [\fIoption\fR] [\fIcommand\fR [\fIarguments\fR]]
.SH DESCRIPTION
\fBlxqt-sudo\fR (and symlinks \fBlxsu\fR, \fBlxsudo\fR) is a graphical QT frontend for plain \fBsudo(8)\fR or \fBsu(1)\fR (for requesting optional password in GUI fashion).
.br
When invoked it simply spawns child \fIsudo\fR or \fIsu\fR process with requested \fIcommand\fR (and optional \fIarguments\fR). If \fIsudo\fR/\fIsu\fR requests user's password,
the GUI password dialog is shown and (after submit) the password is provided to backend.
.br
.SH OPTIONS
\fBoption\fR is one of:
.br
  -h|--help      Print help.
.br
  -v|--version   Print version information.
.br
  -s|--su        Use \fIsu\fR as backend (default for \fBlxqt-sudo\fR & \fBlxsudo\fR is \fIsudo\fR, for \fBlxsu\fR is \fIsu\fR).
.br
  -d|--sudo      Use \fIsudo\fR as backend (default for \fBlxqt-sudo\fR & \fBlxsudo\fR is \fIsudo\fR, for \fBlxsu\fR is \fIsu\fR).
.SH "REPORTING BUGS"
Report bugs to https://github.com/lxqt/lxqt/issues
.SH "SEE ALSO"
\fBsudo(8)\fR \fBsu(1)\fR
.SH AUTHOR
This manual page was created by \fBPalo Kisa\fR \fI<palo.kisa@gmail.com>\fR
for \fBLXQt\fR project.


CHANGELOG

lxqt-build-tools-0.6.0 / 2018-01-25
===================================

  * Bump version to 0.6.0
  * Don't use automatic string conversions
  * Introduced lxqt-transupdate for translation updates
  * Improved cmake scripting
    - Set cmake_minimum_required to 3.1.0
    - Consider QT_SELECT - some distributions use this tool
    - Drop 0X and 11 support, require c++14
    - Remvoed QUIET from any finds
    - Cleaned up LXQtTranslateTs module
    - Removed translation pulling and cleanup external repos
      after translations moved back to their original repos
    - Remove link-time optimization for gcc since it breaks
      qobject_cast() in some cases.
    - Removed obsolete FindFm module

lxqt-build-tools-0.5.0 / 2018-05-21
===================================

  * Bump version to 0.5.0
  * Prevent in-source builds
  * Adds an PreventInSourceBuilds module
  * http --> https
  * Fixed some LXDE mentions
  * LXQtCompilerSettings: Set CMAKE_C_VISIBILITY_PRESET to hidden
  * Don't allow Qt foreach/Q_FOREACH
  * Update CMakeLists.txt
  * Require Qt 5.7.1
  * Adds a FindXKBCommon CMake module
  * Adds ECMFindModuleHelpers CMake module
  * Don't export github templates

0.4.0 / 2017-09-22
==================

  * Release 0.4.0: Update changelog
  * Bump version to 0.4.0
  * Added needed versions
  * remove version set from LXQtConfigVars.in
  * Move LXQt version to liblxqt
  * Fixes a FTBFS with superbuild mode
  * Adds a FindExif module
  * LXQtCompilerSettings: Fixes a typo in diagnostics-color
  * Adds (Ninja) colored diagnostics for GCC (>=4.9)
  * Enable strict iterators for debug builds

0.3.2 / 2017-01-14
==================

  * Release 0.3.2: Update changelog
  * Bump version to 0.3.2 - finally the build tools are arch: all (#15)
  * Warnings (#16)
  * Remove arched bits from version.cmake-file (#14)

0.3.1 / 2017-01-01
==================

  * Release 0.3.1: Update changelog
  * Bump LXQT_PATCH_VERSION and build tools version (#13)
  * 32 64 bit ness (#12)
  * Use "gcc-ar" and "gcc-ranlib" tools when building static libs with gcc link-time optimization. (#11)

0.3.0 / 2016-12-06
==================

  * Release 0.3.0: Update changelog
  * Adds an FindXCB CMake module
  * FindUDev: Add VERSION_STRING support
  * FindUDev: Set REQUIRED_VARS
  * FindUdev: Feed pkg_modules_modules results to find_library/path
  * FindUDev: Adds Feature Summary
  * Adds FindUDev module
  * Enables colored diagnostics for CLang/Ninja combination
  * Disable link-time optimization for clang++/llvm since after some testing it breaks QObject signal/slot sometimes. * Remove trailing spaces.
  * Use different link-time optimization compiler options for gcc and clang and move -Bsymbolic flags to linker flags.
  * Add more linker options to turn on linker-time optimizations for gcc & clang.
  * Set CMAKE_BUILD_TYPE to Release by default if it's value is not set.
  * FindGLIB: Improve gio-unix handling
  * Adds a FindGLIB CMake module
  * Adds a FindMenuCache CMake module
  * Adds a FindFm CMake module
  * move project name to top
  * Fixes the name of package config file name in superbuild mode

0.2.0 / 2016-10-30
==================

  * Release 0.2.0: Update changelog
  * Bump minor to 2 (#7)
  * Added Qt5Core to README.md (#6)
  * Adds an LXQtConfigVars module
  * Adds superbuild support
  * README.md: Update dependencies

0.1.0 / 2016-10-14
==================

  * Release 0.1.0: Update changelog
  * Adds LXQtCreatePortableHeaders module
  * Authors and license (#2)
  * Add README.md
  * Make lxqt-build-tools be usable

0.0.0 / 2016-09-30
==================

  * Remove cpack (#107)
  * Bump version ot 0.11.0 (#105)
  * Settings: Add "homemade" XDG_CONFIG_DIRS support
  * translations: Change default translations repository
  * Don't lock the screen when it's already locked.
  * LXQtTranslateTs.cmake: Enable PULL_TRANSLATIONS by default
  * LXQtTranslateTs.cmake: Use/update template in remote repo
  * .gitignore: Cleanup
  * build: Use external translations
  * translations: Fix external translations logic
  * translations: Add external translations support
  * Adds CMAKE_INSTALL_PREFIX to the LXQt Theme search paths
  * LXQtCompilerSettings: Add QT_USE_QSTRINGBUILDER to the compile definitions
  * TranslateDesktopTs: Adds UPDATE_OPTIONS options
  * TranslateDesktop: Remove lupdate absolute option
  * TranslateDesktop: Use -a switch in grep commands
  * Use CMAKE_INSTALL_FULL_DATAROOTDIR
  * Add LXQT_GRAPHICS_DIR directory definition
  * Adds FindXdgUserDirs CMake find_package module
  * Release 0.10.0
  * pluginTranslation: removed using QStringLiteral as it was causing SEGFAULT in static finalization time
  * pluginTranslation: add some symbol for referencing (for file to be not stripped out in static linking time)
  * Adds "automatic" plugin translations loader support
  * Rename LxQt to LXQt everywhere
  * Handles CMake policy CMP0063
  * Adds an lxqt_enable_target_exceptions() CMake function
  * Adds LXQtCompilerSettings CMake module
  * LXQtCreatePkgConfigFile.cmake: Makes install COMPONENT configurable
  * LXQtTranslateTs.cmake Adds install COMPONENT option
  * Adds Runtime install COMPONENT
  * Fixes an SuperBuild/NormalBuild FTBFS
  * CMake: Use CMAKE_AUTOUIC, drop qt5_wrap()
  * CMake: Quote variables
  * CMake: Removes unused variables
  * Removes LIB_INSTALL_DIR CACHE PATH variable
  * Use relative paths in install DESTINATION
  * Marks link libraries as PUBLIC
  * Use target_compile_definitions() instead of add_definitions()
  * Make Release the default build type
  * Simplifies build type matching
  * lxqt-config.cmake.in: Use find_dependency()
  * Updates the build system to the Targets infrastructure
  * Sort .desktop translations. Fix typos in the comments. Remove local dir artifact.
  * pkg-config: Adds Qt5Xdg to the required dependencies
  * Updates lxqt_create_pkgconfig_file()
  * Stops using Qt5Xdg dropped stuff
  * Adds LXQT_DATA_DIR configuration variable

lxqt-sudo/translations/lxqt-sudo.ts
<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE TS>
<TS version="2.1">
<context>
    <name>PasswordDialog</name>
    <message>
        <location filename="../passworddialog.ui" line="6"/>
        <location filename="../obj-x86_64-linux-gnu/lxqt-sudo_autogen/include/ui_passworddialog.h" line="171"/>
        <source>LXQt sudo</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../passworddialog.ui" line="42"/>
        <location filename="../obj-x86_64-linux-gnu/lxqt-sudo_autogen/include/ui_passworddialog.h" line="173"/>
        <source>Copy command to clipboard</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../passworddialog.ui" line="45"/>
        <location filename="../obj-x86_64-linux-gnu/lxqt-sudo_autogen/include/ui_passworddialog.h" line="175"/>
        <source>&amp;Copy</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../passworddialog.ui" line="83"/>
        <location filename="../obj-x86_64-linux-gnu/lxqt-sudo_autogen/include/ui_passworddialog.h" line="176"/>
        <source>The requested action needs administrative privileges.&lt;br&gt;Please enter your password.</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../passworddialog.ui" line="106"/>
        <location filename="../obj-x86_64-linux-gnu/lxqt-sudo_autogen/include/ui_passworddialog.h" line="178"/>
        <source>LXQt sudo backend</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../passworddialog.ui" line="109"/>
        <location filename="../obj-x86_64-linux-gnu/lxqt-sudo_autogen/include/ui_passworddialog.h" line="181"/>
        <source>A program LXQt sudo calls in background to elevate privileges.</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../passworddialog.ui" line="119"/>
        <location filename="../obj-x86_64-linux-gnu/lxqt-sudo_autogen/include/ui_passworddialog.h" line="183"/>
        <source>Command:</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../passworddialog.ui" line="126"/>
        <location filename="../obj-x86_64-linux-gnu/lxqt-sudo_autogen/include/ui_passworddialog.h" line="184"/>
        <source>Password:</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../passworddialog.ui" line="133"/>
        <location filename="../obj-x86_64-linux-gnu/lxqt-sudo_autogen/include/ui_passworddialog.h" line="186"/>
        <source>Enter password</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../passworddialog.cpp" line="60"/>
        <source>Attempt #%1</source>
        <translation type="unfinished"></translation>
    </message>
</context>
<context>
    <name>QObject</name>
    <message>
        <location filename="../sudo.cpp" line="69"/>
        <source>Usage: %1 option [command [arguments...]]

GUI frontend for %2/%3

Arguments:
  option:
    -h|--help      Print this help.
    -v|--version   Print version information.
    -s|--su        Use %3(1) as backend.
    -d|--sudo      Use %2(8) as backend.
  command          Command to run.
  arguments        Optional arguments for command.

</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../sudo.cpp" line="86"/>
        <source>%1 version %2
</source>
        <translation type="unfinished"></translation>
    </message>
</context>
<context>
    <name>Sudo</name>
    <message>
        <location filename="../sudo.cpp" line="189"/>
        <source>%1: no command to run provided!</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../sudo.cpp" line="196"/>
        <source>%1: no backend chosen!</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../sudo.cpp" line="213"/>
        <source>Syscall error, failed to fork: %1</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../sudo.cpp" line="240"/>
        <source>unset</source>
        <extracomment>shouldn&apos;t be actually used but keep as short as possible in translations just in case.</extracomment>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../sudo.cpp" line="289"/>
        <source>%1: Detected attempt to inject privileged command via LC_ALL env(%2). Exiting!
</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../sudo.cpp" line="331"/>
        <source>Syscall error, failed to bring pty to non-block mode: %1</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../sudo.cpp" line="339"/>
        <source>Syscall error, failed to fdopen pty: %1</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../sudo.cpp" line="308"/>
        <source>%1: Failed to exec &apos;%2&apos;: %3
</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../sudo.cpp" line="370"/>
        <source>Child &apos;%1&apos; process failed!
%2</source>
        <translation type="unfinished"></translation>
    </message>
</context>
</TS>
